language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/lf/EventDecoderGen.scala

```diff
@@ -43,12 +43,12 @@ object EventDecoderGen {
     // the ledger api still uses names with only dots in them, while QualifiedName.toString
     // separates the module and the name in the module with colon.
     def legacyDottedName(identifier: Identifier) = {
-      val Identifier(SimpleString(packageId), QualifiedName(module, name)) = identifier
-      s"${module.dottedName: String}.${name.dottedName: String}@${packageId: String}"
+      val Identifier(packageId, QualifiedName(module, name)) = identifier
+      s"${module.dottedName: String}.${name.dottedName: String}@$packageId"
```

Comment:
it's not paranoia if scala is really out to get you

```suggestion
      s"${module.dottedName: String}.${name.dottedName: String}@${packageId: String}"
```

Reply:
done